src/PEPSync.m

```diff
@@ -347,9 +351,6 @@ - (PEP_STATUS)messageToSendHelper:(struct _message * _Nullable)msg
 - (PEP_STATUS)messageToSend:(struct _message * _Nullable)msg
 {
     PEP_STATUS status = [self messageToSendHelper:msg];
-    if (status == PEP_STATUS_OK) {
```

Comment:
Why this prevents a double free if the point where it happened validates if the msg exists?
Is there a race condition about it?

Reply:
According to your stack trace, the double free is triggered in key_reset.c:1723.  The enc_msg in question has been sent out via send_cb(), which lands in this place in the objc wrapper and already has been freed here.

No race condition involved, and I don't get the reference to "validation if the msg exists".